rules-tests/CodeQuality/Rector/Ternary/TernaryEmptyArrayArrayDimFetchToCoalesceRector/Fixture/some_array_key.php.inc

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rector\Tests\CodeQuality\Rector\Ternary\TernaryEmptyArrayArrayDimFetchToCoalesceRector\Fixture;
+
+final class SomeArrayKey
+{
+    private $items;
+
+    public function run()
+    {
+        return ! empty($this->items[0]) ? $this->items[0] : 'default';
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\CodeQuality\Rector\Ternary\TernaryEmptyArrayArrayDimFetchToCoalesceRector\Fixture;
+
+final class SomeArrayKey
+{
+    private $items;
+
+    public function run()
+    {
+        return $this->items[0] ?: 'default';
```

Comment:
This will emit a warning, while the code before transformation does not

https://3v4l.org/I6Rb7

Reply:
Indeed, thanks for checking

Reply:
or empty array https://3v4l.org/IrSjv , it seems `??` only works when it nullable.